core/src/prim/bytes.rs

```diff
@@ -134,10 +134,12 @@ impl BytesOp {
         let x = xs.last();
         x.map(|x| U8(*x))
       }
-      (Self::Tail, Bytes(xs)) => {
+      (Self::Tail, Bytes(xs)) => if xs.is_empty() {
+        None
```

Comment:
Oh hang on, in that case, should this also return an empty Bytes([])?

Reply:
Yes, that's how `tail` works in e.g. Haskell too

Reply:
Ok, I've amended that too. Hopefully there's no other changes required, but if not, please let me know!

Reply:
Oh hang I think I need to do the same for BitsOp::Tail.